pinot-query-planner/src/test/resources/queries/LiteralEvaluationPlans.json

```diff
@@ -0,0 +1,161 @@
+        "sql": "EXPLAIN PLAN FOR SELECT dateTrunc('MONTH', 854755200000) AS day FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(day=[CAST(854755200000:BIGINT):BIGINT])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "Eval Filter literal function",
+        "sql": "EXPLAIN PLAN FOR SELECT * FROM a WHERE  dateTrunc('MONTH', ts) =  dateTrunc('MONTH', FROMDATETIME('2019-01-01 00:00:00', 'yyyy-MM-dd HH:mm:ss'))",
+        "output": [
+          "Execution Plan",
+          "\nLogicalFilter(condition=[=(DATETRUNC('MONTH', $6), 1546300800000)])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "upper case literal function",
+        "sql": "EXPLAIN PLAN FOR SELECT upper('month') FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=['MONTH'])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "concat literal function",
+        "sql": "EXPLAIN PLAN FOR SELECT concat('month', ' 1') FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=[CAST('month 1':VARCHAR):VARCHAR])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "substr literal function",
+        "sql": "EXPLAIN PLAN FOR SELECT substr('month',2) FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=[CAST('nth':VARCHAR):VARCHAR])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "chained string functions on literal",
+        "sql": "EXPLAIN PLAN FOR SELECT upper(lower(upper(substr('month',2)))) FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=[CAST('NTH':VARCHAR):VARCHAR])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "geoToH3 functions",
+        "sql": "EXPLAIN PLAN FOR Select geoToH3(ST_point(20,102),5) FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=[599041711439609855:BIGINT])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "geoToH3 functions",
+        "sql": "EXPLAIN PLAN FOR Select ST_Point(20, 10, 1) FROM a",
+        "output": [
+          "Execution Plan",
+          "\nLogicalProject(EXPR$0=[X'8040340000000000004024000000000000':VARBINARY])",
+          "\n  LogicalTableScan(table=[[a]])",
+          "\n"
+        ]
+      },
+      {
+        "description": "St distance with two bytes literals",
+        "sql": "EXPLAIN PLAN FOR Select ST_Distance(X'8040340000000000004024000000000000', ST_Point(-122, 37.5, 1)) FROM a",
```

Comment:
This is not PostgreSQL format. Does the current parser support PostgreSQL format? https://www.postgresql.org/docs/current/datatype-binary.html

Reply:
No, current parser doesn't support it. We need to support PostgreSQL format later.